src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -312,7 +312,7 @@ static CodeTypeDeclaration CreateTypeDeclaration(TypeDefinition publicType, stri
                 }
             }
 
-            return declaration;
+            return declaration.Sort();
         }
 
         static CodeTypeDeclaration CreateDelegateDeclaration(TypeDefinition publicType, AttributeFilter attributeFilter)
```

Comment:
`CreateDelegateDeclaration` creates a special type of `CodeTypeDeclaration` which is of type `CodeTypeDelegate`. I think it would make sense for consistency reason to also sort parameters on delegate declarations. What do you think?

```
                var sorted= new CodeTypeDelegate(@delegate.Name)
                {
                    Attributes = @delegate.Attributes,
                    CustomAttributes = @delegate.CustomAttributes,
                    ReturnType = @delegate.ReturnType,
                };
                sorted.Parameters.AddRange(@delegate.Parameters);
```

Reply:
OK

Reply:
I'm not sure I understand. I'm not sure there's anything to sort in a delegate. Does your code snippet actually change anything?

Reply:
@adamralph your code also sorts by the parameters of a method right? See https://github.com/ApiApprover/ApiApprover/pull/138/files#diff-9cab084db4a89e617009fb6111b48f7eR50-R53

But I guess you are implicitly saying that even though a delegate has parameters it doesn't affect any sorting because it is sort of stand alone?

Or would having a delegate as part of a declaration change anything? And that is why I'm trying to explore if delegate type declarations should also be taken into account in the sorting.

Reply:
@danielmarbach a delegate doesn't have members, i..e this is a delegate declaration _in it's entirety_:

```c#
public delegate void Foo();
```

So there is nothing to sort within a `CodeTypeDeclaration` (if undertsand the type correctly) which represents a delegate.

We could sort delegates _within a namespace_ by their type parameters and parameters, similar to how this PR is doing for members _within a type_, but I think that would be best handled elsewhere, because it would form part of how all types (classes, structs, interfaces, delegates, enums, etc.) are sorted within a namespace. I'd be happy to investigate that as well, but I think it should be a seperate PR.